src/components/dropdown/dropdown.svelte

```diff
@@ -83,7 +86,13 @@
     bind:isOpen
     bind:currentValue={value}
     on:select-item={onItemSelect}
-    on:close={() => button.focus()}
+    on:close={(e) => {
+      if (e.detail.originalEvent.composedPath().includes(dropdown)) {
+        e.preventDefault()
+      } else {
+        button.focus()
+      }
```

Comment:
I think it's fine but, again, needs a comment especially here about why we need to do it